device/transport/amqp/devdoc/device_amqp_requirements.md

```diff
@@ -109,6 +109,25 @@ The `sendEvent` method sends an event to the IoT Hub as the device indicated in
 
 **SRS_NODE_DEVICE_AMQP_16_052: [** The `sendEventBatch` method shall throw a `NotImplementedError`. **]**
 
+### sendOutputEvent(outputName: string, message: Message, done: (err?: Error, result?: results.MessageEnqueued) => void): void;
+The `sendOutputEvent` method sends an event to the IoT Hub as the device indicated in the constructor argument.
+
+**SRS_NODE_DEVICE_AMQP_18_005: [** The `sendOutputEvent` method shall connect and authenticate the transport if necessary. **]**
+
+**SRS_NODE_DEVICE_AMQP_18_006: [** The `sendOutputEvent` method shall create and attach the d2c link if necessary. **]**
+
+**SRS_NODE_DEVICE_AMQP_18_007: [** The `sendOutputEvent` method shall construct an AMQP request using the message passed in argument as the body of the message. **]**
+
+**SRS_NODE_DEVICE_AMQP_18_012: [** The `sendOutputEvent` method  shall set the annotation "x-opt-output-name" on the message to the `outputName`. **]**
+
+**SRS_NODE_DEVICE_AMQP_18_008: [** The `sendOutputEvent` method shall call the `done` callback with a null error object and a MessageEnqueued result object when the message has been successfully sent. **]**
+
+**SRS_NODE_DEVICE_AMQP_18_009: [** If `sendOutputEvent` encounters an error before it can send the request, it shall invoke the `done` callback function and pass the standard JavaScript Error object with a text description of the error (err.message). **]**
+
+### sendOutputEventBatch(outputName: string, messages: Message[], done: (err?: Error, result?: results.MessageEnqueued) => void): void;
+
+**SRS_NODE_DEVICE_AMQP_18_004: [** `sendOutputEventBatch` shall throw a `NotImplementedError`. **]**
```

Comment:
this might be more important for modules than it is for devices. We should carve out time to figure out message batching in AMQP.

Reply:
added 2366357

---
In reply to: [180588896](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r180588896) [](ancestors = 180588896)